flake-module.nix

```diff
@@ -0,0 +1,176 @@
+                };
+
+                haskellPackages = lib.mkOption {
+                    description = ''
+                        List of Haskell packages to be installed in the IHP environment.
+                    '';
+                    default = p: [
+                        p.cabal-install
+                        p.base
+                        p.wai
+                        p.text
+                        p.hlint
+                        p.ihp
+                    ];
+                };
+
+                projectPath = lib.mkOption {
+                    description = ''
+                        Path to the IHP project. You likely want to set this to `./.`.
+                    '';
+                    type = lib.types.path;
+                };
+
+                dontCheckPackages = lib.mkOption {
+                    description = ''
+                        List of Haskell package names whose tests are skipped during build.
+                    '';
+                    type = lib.types.listOf (lib.types.str);
+                    default = [];
+                };
+
+                doJailbreakPackages = lib.mkOption {
+                    description = ''
+                        List of Haskell package names who are jailbreaked before build.
+                    '';
+                    type = lib.types.listOf (lib.types.str);
+                    default = [];
+                };
+
+                dontHaddockPackages = lib.mkOption {
+                    description = ''
+                        List of Haskell package names whose haddock is not built during app build.
+                    '';
+                    type = lib.types.listOf (lib.types.str);
+                    default = [];
+                };
+            };
+        }
+    );
+
+    config = {
+        perSystem = { self', lib, pkgs, system, config, ... }: let
+            cfg = config.ihp;
+            ihp = ihpFlake.inputs.self;
+            ghcCompiler = import "${ihp}/NixSupport/mkGhcCompiler.nix" {
+                inherit pkgs;
+                inherit (cfg) ghcCompiler dontCheckPackages doJailbreakPackages dontHaddockPackages;
+                ihp = ihp;
+                haskellPackagesDir = cfg.projectPath + "/Config/nix/haskell-packages";
+            };
+        in lib.mkIf cfg.enable {
+            _module.args.pkgs = import ihpFlake.inputs.nixpkgs { inherit system; };
+
+            # release build package
+            packages.default = import "${ihp}/NixSupport/default.nix" {
+                ihp = ihp;
+                haskellDeps = cfg.haskellPackages;
+                otherDeps = p: cfg.packages;
+                projectPath = cfg.projectPath;
+
+                # Dev tools are not needed in the release build
+                includeDevTools = false;
+
+                # Set optimized = true to get more optimized binaries, but slower build times
+                # TODO make configurable via option
+                optimized = false;
```

Comment:
The old Make system has two targets to archive this:

```
make build/bin/RunOptimizedProdServer # Slow build, optimized binary
make build/bin/RunUnoptimizedProdServer # Fast build, no optimizations
```

Can we do something similiar here? Like two output binaries in our flake, so the end user can decide to get either an optimized or unoptimized build?

Reply:
Makes sense, we should be able to provide two flake outputs for optimized/unoptimized builds, so you can build them with something like `nix build .#optimizedServer` or `nix build .#unoptimizedServer` and additionally provide an optional option that controls what is built as default package, i.e. what's build with a plain `nix build`. I'll figure something out.

Reply:
We should likely add these nix build commands to the documentation later on